ast/ast.go

```diff
@@ -31,8 +34,11 @@ type BooleanExpr interface {
 
 // FieldExpr is the interface implemented by expressions that reference fields.
 type FieldExpr interface {
-	fieldExprNode()
-	Copy() FieldExpr
+	// AsString() generates the original zql source for a FieldExpr.
```

Comment:
I was about to add a comment like "hmm maybe we can do this differently later" but now I don't see it used anywhere. Do we have a need for it coming up?

Reply:
Oh I see now, we do have a need for it. Hmm, could it be moved somewhere else?  Seems a bit specialized. Just wondering and can also be done later.

Reply:
Yeah, good point, I'll move this elsewhere.